src/lib/snack-bar/snack-bar.ts

```diff
@@ -45,7 +46,7 @@ export function MAT_SNACK_BAR_DEFAULT_OPTIONS_FACTORY(): MatSnackBarConfig {
  * Service to dispatch Material Design snack bar messages.
  */
 @Injectable({providedIn: MatSnackBarModule})
-export class MatSnackBar {
+export class MatSnackBar implements OnDestroy {
```

Comment:
Since `@Injectable`s are just regular classes that are used as singletons, I don't believe there are any lifecycle hooks.

A quick test of checking for usage of `OnInit` shows that it is never called.

Reply:
AFAIK they only have `OnDestroy`. I believe it gets called when the associated module is unloaded. We've been doing it for the other services as well.

Reply:
Ahh, interesting.  Didn't know about that.  Then this makes sense to do :+1: